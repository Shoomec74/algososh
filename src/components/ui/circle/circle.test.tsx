import React from "react";
import { render } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Circle", () => {
  it("renders without a letter", () => {
    const { container } = render(<Circle />);
    expect(container).toMatchSnapshot();
  });

  it("renders with a letter", () => {
    const { container } = render(<Circle letter="A" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with head", () => {
    const { container } = render(<Circle head="text" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with a react element in head", () => {
    const { container } = render(<Circle head={<div>React element</div>} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with tail", () => {
    const { container } = render(<Circle tail="text" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with a react element in tail", () => {
    const { container } = render(<Circle tail={<div>React element</div>} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with index", () => {
    const { container } = render(<Circle index={1} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with isSmall prop set to true", () => {
    const { container } = render(<Circle isSmall />);
    expect(container).toMatchSnapshot();
  });

  it("renders in default state", () => {
    const { container } = render(<Circle state={ElementStates.Default} />);
    expect(container).toMatchSnapshot();
  });

  it("renders in changing state", () => {
    const { container } = render(<Circle state={ElementStates.Changing} />);
    expect(container).toMatchSnapshot();
  });

  it("renders in modified state", () => {
    const { container } = render(<Circle state={ElementStates.Modified} />);
    expect(container).toMatchSnapshot();
  });
});
