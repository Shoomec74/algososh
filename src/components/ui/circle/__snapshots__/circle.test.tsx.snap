// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle renders in changing state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle renders in default state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle renders in modified state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle renders with a letter 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle renders with a react element in head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div>
        React element
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle renders with a react element in tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div>
        React element
      </div>
    </div>
  </div>
</div>
`;

exports[`Circle renders with head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      text
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle renders with index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      1
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</div>
`;

exports[`Circle renders with isSmall prop set to true 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Circle renders with tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      text
    </div>
  </div>
</div>
`;

exports[`Circle renders without a letter 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;
